### PR TITLE
fix: enable strong session protection by default

### PR DESCRIPTION
**Pull request by dpgaspar**
### SUMMARY

Enable strong session protection by default. This is a more secure sane default, more details on: https://flask-login.readthedocs.io/en/latest/#session-protection

> When session protection is active, each request, it generates an identifier for the user’s computer (basically, a secure hash of the IP address and user agent). If the session does not have an associated identifier, the one generated will be stored. If it has an identifier, and it matches the one generated, then the request is OK.
If the identifiers do not match in basic mode, or when the session is permanent, then the session will simply be marked as non-fresh, and anything requiring a fresh login will force the user to re-authenticate. (Of course, you must be already using fresh logins where appropriate for this to have an effect.)
If the identifiers do not match in strong mode for a non-permanent session, then the entire session (as well as the remember token if it exists) is deleted.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/config.py [hunk 1]
```diff
@@ -1387,6 +1387,8 @@ def EMAIL_HEADER_MUTATOR(  # pylint: disable=invalid-name,unused-argument
 SESSION_COOKIE_HTTPONLY = True  # Prevent cookie from being read by frontend JS?
 SESSION_COOKIE_SECURE = False  # Prevent cookie from being transmitted over non-tls?
 SESSION_COOKIE_SAMESITE: Optional[Literal["None", "Lax", "Strict"]] = "Lax"
+# Accepts None, "basic" and "strong", more details on: https://flask-login.readthedocs.io/en/latest/#session-protection
+SESSION_PROTECTION = "strong"
```
Comment (michael-s-molina) [inline superset/config.py:1391]: Do you think it's worth adding something to `UPDATING.md`?

Comment (dpgaspar) [reply superset/config.py:1391]: I think so, more is better then less

**Review by michael-s-molina — APPROVED** [review]
LGTM

**Review by villebro — APPROVED** [review]
LGTM

**Comment by virgofx** [comment]
Just an FYI - With session protection strong, this breaks selenium drivers connecting to app. I had to revert to none which worked but didn't test basic. There may be some additional work here to update the machine auth function to properly work when this is enabled

**Comment by mapledan** [comment]
I am facing a same problem where the alert and report are not working properly. 
To resolve this issue, I need to configure the SESSION_PROTECTION to None.

> Just an FYI - With session protection strong, this breaks selenium drivers connecting to app (which means things like cache warmup and thumbnails don't work). I had to revert to none which worked but didn't test basic. There may be some additional work here to update the machine auth function to properly work when this is enabled

**Comment by sfirke** [comment]
Alerts & Reports works for me with `SESSION_PROTECTION = "basic"`.

`"strong"` prevented me from logging in entirely, so I couldn't test Alerts & Reports with that value.

**Comment by dpgaspar** [comment]
ok, thank you for the reports, I'll work on a fix for this

**Pull request merged**